### ISSUE TITLE
Free tier performance is incredibly slow.

### ISSUE THREAD

**Issue by KelSolaar** [microsoft/AzureNotebooks#816]
TLDR Summary:

**Google Colab Free Tier was x41 faster than Azure Notebooks Free Tier!**

|           | *Azure*   | *Colab*  | *Factor* |
|-----------|-----------|----------|----------|
| Setup     | 594 s     | 31 s     | x19.2    |
| Cloning   | 500 KiB/s | 30 MiB/s | x61.4    |
| Import    | 186 s     | 2.7 s    | x68.9    |
| Execution | 188 s     | 13 s     | x14.5    |

Hi,

I'm gobsmacked by how slow the free Azure Notebooks tier is, it is incredibly painful to use, well, actually almost unusable which really came as an extremely disappointing surprise.

Given those Notebooks:

- https://notebooks.azure.com/kelsolaar-fool/projects/gamut-mapping-ramblings/html/gamut_mapping_ramblings_01.ipynb
- https://colab.research.google.com/drive/1h9v5hYbx0W7eJVVw6GDsuPRYHClt0JYM?usp=sharing (Compare Time with Azure Revision)

1. The initial setup takes 594 s on Azure vs 31 s on Colab (even though Azure had already the principal package cached from my first test and could not find a way to completely reset the instance which I did in Colab)
2. Repo cloning speed was about 500 KiB/s on Azure vs 30 MiB/s on Colab
3. Import time is 186 s on Azure vs 2.7 s on Colab (WOW!)
4. Execution time is 188 s on Azure vs 13 s on Colab

I should have run those tests multiple times and given the average/mean. That being said, having restarted the kernel a few times on Azure, I did not notice any practical difference; nothing that would close the x41 speed gap anyway.

The obvious issue with providing a free tier with that level of performance is that it is actually doing a disservice for the paid product. I sincerely hope it was just a transient problem.
 
#### Which site did you access us from?
    [x] https://notebooks.azure.com
    [] https://studio.azureml.net

#### What steps can we take to reproduce this issue? If you have this in a notebook already, you can just share a link here.

Simply run the shared notebook.

#### Which language/kernel were you using when you experienced this issue?
    [] Python 2.7 
    [x] Python 3.5
    [] Python 3.6 
    [] R 
    [] F#
    [] Terminal

```
===============================================================================
*                                                                             *
*   Interpreter :                                                             *
*       python : 3.5.4 |Anaconda custom (64-bit)| (default, Oct 13 2017,      *
*   11:22:58)                                                                 *
*                [GCC 7.2.0]                                                  *
*                                                                             *
*   colour-science.org :                                                      *
*       colour : v0.3.15-124-ge226e02                                         *
*                                                                             *
*   Runtime :                                                                 *
*       imageio : 2.6.1                                                       *
*       matplotlib : 2.1.1                                                    *
*       networkx : 2.4                                                        *
*       numpy : 1.17.3                                                        *
*       pandas : 0.19.2                                                       *
*       scipy : 1.1.0                                                         *
*       six : 1.11.0                                                          *
*                                                                             *
===============================================================================
```

#### What time (please include time zone) did this occur?
Pacific/Auckland (+12)

#### Which Web Browser(s) did you see the error in?
    [x] Chrome
    [] Firefox
    [] Internet Explorer / Edge (Please note this is not supported)
    [] Other (Please Specify)

#### Which Operating System are you using? Please specify version as well if possible:
    [] Windows
    [x] MacOS
    [] Ubuntu
    [] Other (Please Specify)

#### What is the path to the notebook file (if relevant) that reproduces the issue? For example, https://notebooks.azure.com/<USER-NAME>/libraries/<LIBRARY-NAME>/html/<PATH-TO-FILE>. If this is sensitive, you can email it to nbhelp@microsoft.com

https://notebooks.azure.com/kelsolaar-fool/projects/gamut-mapping-ramblings/html/gamut_mapping_ramblings_01.ipynb
    
#### Each notebook instance has log files. It may be helpful for us to have them. You can retrieve them by following this [help article](https://notebooks.azure.com/help/jupyter-notebooks/troubleshooting/jupyter-debug-log)

**Comment by KelSolaar** [comment]
I can actually see a similar issue here reported in 2018: https://github.com/microsoft/AzureNotebooks/issues/400

**Comment by kevglynn** [comment]
I have to bandwagon the OP's sentiment. A few weeks ago, it was usable. In the last week or so, there's a new issue. On top of slow processing, there is also an incredibly high latency: lag lag lag, waiting for the output to update on my screen.

**Comment by ahnitz** [comment]
I'll have to add that I'm seeing the same issue. We had planned to use this for an online webinar (azure notebooks advertises itself as being good for this!), but I'm now moving our materials to an alternative. Unfortunately, I cannot recommend Microsoft Azure at the current performance level for educational/class uses. It used to be quite reasonable for this, which is why the drastic changes is such a shame.

**Comment by dhiraj81** [comment]
+1 The performance of AzureML is super slow. I would rather run Python notebooks on local machine than using AzureML.
